packages/alfa-html/src/alphabet.ts

```diff
@@ -237,6 +279,522 @@ const tagName: Pattern = (stream, emit, state) => {
+
+    case Char.LessThanSign:
+      emit({ type: TokenType.Character, data: "<" });
+      return scriptDataDoubleEscapedLessThanSign;
+
+    case Char.Null:
+      emit({ type: TokenType.Character, data: "\ufffd" });
+      return scriptDataDoubleEscaped;
+
+    case null:
+      return Command.End;
+
+    default:
+      emit({ type: TokenType.Character, data: fromCharCode(char) });
+      return scriptDataDoubleEscaped;
+  }
+};
+
+/**
+ * 8.2.4.29
+ * @see https://www.w3.org/TR/html/syntax.html#tokenizer-script-data-double-escaped-dash-dash-state
+ */
+const scriptDataDoubleEscapedDashDash: Pattern = (stream, emit, state) => {
+  const char = stream.next();
+
+  switch (char) {
+    case Char.HyphenMinus:
+      emit({ type: TokenType.Character, data: "-" });
+      break;
+
+    case Char.LessThanSign:
+      emit({ type: TokenType.Character, data: "<" });
+      return scriptDataDoubleEscapedLessThanSign;
+
+    case Char.GreaterThanSign:
+      emit({ type: TokenType.Character, data: ">" });
+      return scriptData;
+
+    case Char.Null:
+      emit({ type: TokenType.Character, data: "\ufffd" });
+      return scriptDataDoubleEscaped;
+
+    case null:
+      return Command.End;
+
+    default:
+      emit({ type: TokenType.Character, data: fromCharCode(char) });
+      return scriptDataDoubleEscaped;
+  }
+};
+
+/**
+ * 8.2.4.30
+ * @see https://www.w3.org/TR/html/syntax.html#tokenizer-script-data-double-escaped-less-than-sign-state
+ */
+const scriptDataDoubleEscapedLessThanSign: Pattern = (stream, emit, state) => {
+  const char = stream.peek(0);
+
+  if (char === Char.Solidus) {
+    stream.advance(1);
+    state.temporaryBuffer = "";
+    emit({ type: TokenType.Character, data: "/" });
+    return scriptDataDoubleEscapeEnd;
+  }
+
+  return scriptDataDoubleEscaped;
+};
+
+/**
+ * 8.2.4.31
+ * @see https://www.w3.org/TR/html/syntax.html#script-data-double-escape-end-state
+ */
+const scriptDataDoubleEscapeEnd: Pattern = (stream, emit, state) => {
+  const char = stream.peek(0);
+
+  if (char === null) {
+    return scriptDataDoubleEscaped;
+  }
+
+  switch (char) {
+    case Char.CharacterTabulation:
+    case Char.LineFeed:
+    case Char.FormFeed:
+    case Char.Space:
+    case Char.Solidus:
+    case Char.GreaterThanSign:
+      stream.advance(1);
+      if (state.temporaryBuffer === "script") {
+        return scriptDataEscaped;
+      }
+
+      emit({ type: TokenType.Character, data: fromCharCode(char) });
+      return scriptDataDoubleEscaped;
+
+    default:
+      if (isUpperCase(char)) {
+        stream.advance(1);
+        state.temporaryBuffer += fromCharCode(char).toLowerCase();
+        emit({ type: TokenType.Character, data: fromCharCode(char) });
+        break;
+      }
+
+      if (isLowerCase(char)) {
+        stream.advance(1);
+        state.temporaryBuffer += fromCharCode(char);
+        emit({ type: TokenType.Character, data: fromCharCode(char) });
+        break;
+      }
```

Comment:
Same as previously; just check for an alphabetic char and lowercase indiscriminately.